abstract_domains/numerical/numerical.py

```diff
@@ -0,0 +1,61 @@
+from abc import ABCMeta, abstractmethod
+from core.expressions import VariableIdentifier, Expression, UnaryArithmeticOperation
+
+
+# rename (without lattice) e.g. IntegerOperations
+class NumericalLattice(metaclass=ABCMeta):
```

Comment:
Isn't this class representing sets of values and the operations on them? I would maybe rename this class NumericalValue or just Value. The class Interval would inherit from it. Alternatively, we just drop this class entirely and just keep the class Interval.